### PR TITLE
Remove separation environment

### PR DESCRIPTION
**Pull request by ethax-ross**
### Context

Now that we have separated NPQ from ECF we want to decommission the separation environment.

### Changes proposed in this pull request

- Remove separation environment

Remove all terraform, Rails environment, GitHub workflows, Makefile commands, DB config and readme references to the separation environment as we are tearing it down.

### Guidance for review

We should be able to check out this branch then run `make ci migration terraform-destroy` - once destroyed we should drop the temp commit before merging.

Tested with `terraform plan -destroy`:

```
DOCKER_IMAGE=ghcr.io/dfe-digital/early-careers-framework:cd0a034d0ef9f756715a1f07a0911d9585ba2e8c make ci separation terraform-plan-destroy

...

Plan: 0 to add, 0 to change, 18 to destroy.
```

### CODE REVIEW

File: app/services/valid_test_data_generators/separation_shared_data.rb [hunk 1]
```diff
@@ -1,66 +0,0 @@
-# frozen_string_literal: true
-
-module ValidTestDataGenerators
-  class SeparationSharedData < ECFLeadProviderPopulater
```
Comment (cwrw) [inline app/services/valid_test_data_generators/separation_shared_data.rb:4]: can we change this instead to sandbox, we might need this later if we ever rerun that data?

File: db/data/separation_shared_data.yml [hunk 2]
```diff
@@ -1,125 +0,0 @@
----
-Best Practice Network:
```
Comment (cwrw) [inline db/data/separation_shared_data.yml:2]: same here, can we change this to sandbox?

File: lib/tasks/separation_data.rake [hunk 3]
```diff
@@ -1,22 +0,0 @@
...
-    return unless Rails.env.in?(%w[development review separation])
-
-    lead_provider = LeadProvider.find_by(name: args[:lead_provider_name])
-    raise "LeadProvider not found: #{args[:lead_provider_name]}" if args[:lead_provider_name] && !lead_provider
-
-    cohort = Cohort.find_by(start_year: args[:cohort_start_year])
-    raise "Cohort not found: #{args[:cohort_start_year]}" if args[:cohort_start_year] && !cohort
-
-    Array.wrap(lead_provider || LeadProvider.name_order.joins(:cohorts).includes(:cohorts)).each do |lp|
-      Array.wrap(cohort || Cohort.between_years((Date.current - 2.years + 1.day).year, (Date.current + 1.year).year)).each do |c|
```
Comment (cwrw) [inline lib/tasks/separation_data.rake:15]: could we change this to sandbox?

File: spec/services/valid_test_data_generators/separation_shared_data_spec.rb [hunk 4]
```diff
@@ -1,55 +0,0 @@
-# frozen_string_literal: true
-
-require "rails_helper"
-
-RSpec.describe ValidTestDataGenerators::SeparationSharedData do
```
Comment (cwrw) [inline spec/services/valid_test_data_generators/separation_shared_data_spec.rb:5]: same here :)

**Review by cwrw — COMMENTED** [review]
looks good some minor comments

**Review by leandroalemao — APPROVED** [review]

**Pull request merged**